test-app/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationActivity.kt

```diff
@@ -0,0 +1,409 @@
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.internal.formatter.MapboxDistanceFormatter
+import com.mapbox.navigation.core.replay.MapboxReplayer
+import com.mapbox.navigation.core.replay.ReplayLocationEngine
+import com.mapbox.navigation.core.replay.route.ReplayProgressObserver
+import com.mapbox.navigation.core.replay.route.ReplayRouteMapper
+import com.mapbox.navigation.core.trip.session.BannerInstructionsObserver
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.examples.util.RouteLine
+import com.mapbox.navigation.examples.util.Utils
+import com.mapbox.navigation.ui.base.api.maneuver.ManeuverCallback
+import com.mapbox.navigation.ui.base.api.maneuver.StepDistanceRemainingCallback
+import com.mapbox.navigation.ui.base.api.maneuver.UpcomingManeuversCallback
+import com.mapbox.navigation.ui.base.model.maneuver.ManeuverState
+import com.mapbox.navigation.ui.base.model.tripprogress.DistanceRemainingFormatter
+import com.mapbox.navigation.ui.base.model.tripprogress.EstimatedTimeToArrivalFormatter
+import com.mapbox.navigation.ui.base.model.tripprogress.PercentDistanceTraveledFormatter
+import com.mapbox.navigation.ui.base.model.tripprogress.TimeRemainingFormatter
+import com.mapbox.navigation.ui.base.model.tripprogress.TripProgressUpdateFormatter
+import com.mapbox.navigation.ui.maneuver.api.MapboxManeuverApi
+import com.mapbox.navigation.ui.maps.camera.NavigationCamera
+import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
+import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSourceOptions
+import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
+import com.mapbox.navigation.ui.tripprogress.api.MapboxTripProgressApi
+import com.mapbox.navigation.ui.utils.internal.ifNonNull
+import kotlinx.android.synthetic.main.layout_activity_navigation.*
+
+class MapboxNavigationActivity :
+    AppCompatActivity(),
+    OnMapLongClickListener {
+
+    private lateinit var mapboxMap: MapboxMap
+    private lateinit var directionsRoute: DirectionsRoute
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var locationComponent: LocationComponentPlugin
+    private lateinit var navigationCamera: NavigationCamera
+    private lateinit var viewportDataSource: MapboxNavigationViewportDataSource
+    private lateinit var tripProgressApi: MapboxTripProgressApi
+    private lateinit var maneuverApi: MapboxManeuverApi
+
+    private val mapboxReplayer = MapboxReplayer()
+    private val replayRouteMapper = ReplayRouteMapper()
+    private val slackLine = RouteLine(this)
```

Comment:
This activity shouldn't use any external classes. Could you set up route line API directly in the activity?